modules/store/src/provide_store.ts

```diff
@@ -0,0 +1,343 @@
+  config?: StoreConfig<T, V> | InjectionToken<StoreConfig<T, V>>
+): ImportedNgModuleProviders;
+export function provideState<T, V extends Action = Action>(
+  slice: FeatureSlice<T, V>
+): ImportedNgModuleProviders;
+export function provideState(
+  featureNameOrSlice: string | FeatureSlice<any, any>,
+  reducers?:
+    | ActionReducerMap<any, any>
+    | InjectionToken<ActionReducerMap<any, any>>
+    | ActionReducer<any, any>
+    | InjectionToken<ActionReducer<any, any>>,
+  config: StoreConfig<any, any> | InjectionToken<StoreConfig<any, any>> = {}
+): ImportedNgModuleProviders {
+  return {
+    ɵproviders: [
+      ..._provideState(featureNameOrSlice, reducers, config),
+      ENVIRONMENT_STATE_PROVIDER,
+    ],
+  };
+}
+
+export function _provideStore(
+  reducers:
+    | ActionReducerMap<any, any>
+    | InjectionToken<ActionReducerMap<any, any>>,
+  config: RootStoreConfig<any, any>
+) {
+  return [
+    {
+      provide: _ROOT_STORE_GUARD,
+      useFactory: _provideForRootGuard,
+      deps: [[Store, new Optional(), new SkipSelf()]],
+    },
+    { provide: _INITIAL_STATE, useValue: config.initialState },
+    {
+      provide: INITIAL_STATE,
+      useFactory: _initialStateFactory,
+      deps: [_INITIAL_STATE],
+    },
+    { provide: _INITIAL_REDUCERS, useValue: reducers },
+    {
+      provide: _STORE_REDUCERS,
+      useExisting:
+        reducers instanceof InjectionToken ? reducers : _INITIAL_REDUCERS,
+    },
+    {
+      provide: INITIAL_REDUCERS,
+      deps: [Injector, _INITIAL_REDUCERS, [new Inject(_STORE_REDUCERS)]],
+      useFactory: _createStoreReducers,
+    },
+    {
+      provide: USER_PROVIDED_META_REDUCERS,
+      useValue: config.metaReducers ? config.metaReducers : [],
+    },
+    {
+      provide: _RESOLVED_META_REDUCERS,
+      deps: [META_REDUCERS, USER_PROVIDED_META_REDUCERS],
+      useFactory: _concatMetaReducers,
+    },
+    {
+      provide: _REDUCER_FACTORY,
+      useValue: config.reducerFactory ? config.reducerFactory : combineReducers,
+    },
+    {
+      provide: REDUCER_FACTORY,
+      deps: [_REDUCER_FACTORY, _RESOLVED_META_REDUCERS],
+      useFactory: createReducerFactory,
+    },
+    ACTIONS_SUBJECT_PROVIDERS,
+    REDUCER_MANAGER_PROVIDERS,
+    SCANNED_ACTIONS_SUBJECT_PROVIDERS,
+    STATE_PROVIDERS,
+    STORE_PROVIDERS,
+    provideRuntimeChecks(config.runtimeChecks),
+    checkForActionTypeUniqueness(),
+  ];
+}
+
+/**
+ * Environment Initializer used in the root
+ * providers to initialize the Store
+ */
+const ENVIRONMENT_STORE_PROVIDER = [
+  {
+    provide: ENVIRONMENT_INITIALIZER,
+    multi: true,
+    useFactory() {
+      return () => {
+        inject(ActionsSubject);
+        inject(ReducerObservable);
+        inject(ScannedActionsSubject);
+        inject(Store);
+        inject(_ROOT_STORE_GUARD, InjectFlags.Optional);
+        inject(_ACTION_TYPE_UNIQUENESS_CHECK, InjectFlags.Optional);
+      };
+    },
+  },
+];
```

Comment:
Should we also inject the root provider to the `FEATURE_STATE_PROVIDER` factory to ensure that `provideStore` logic is executed before `provideState`?

```suggestion
export const ROOT_STORE_PROVIDER = new InjectionToken<void>('@ngrx/store Root Store Provider');

function rootStoreProviderFactory(): void {
  inject(ActionsSubject);
  inject(ReducerObservable);
  inject(ScannedActionsSubject);
  inject(Store);
  inject(_ROOT_STORE_GUARD, InjectFlags.Optional);
  inject(_ACTION_TYPE_UNIQUENESS_CHECK, InjectFlags.Optional);
}

const ENVIRONMENT_STORE_PROVIDER: Provider[] = [
  { provide: ROOT_STORE_PROVIDER, useFactory: rootStoreProviderFactory },
  {
    provide: ENVIRONMENT_INITIALIZER,
    multi: true,
    useFactory() {
      return () => inject(ROOT_STORE_PROVIDER);
    },
  },
];
```

Then we can inject the `ROOT_STORE_PROVIDER` token in `featurStateProviderFactory` to ensure that the root logic is executed:

```ts
function featureStateProviderFactory(): void {
  inject(ROOT_STORE_PROVIDER);

  // register store feature
}
```

We'll also need the `ROOT_STORE_PROVIDER` token for `provideEffects` function.

Reply:
Yep, added